lib/gnat/jetstream/api/object.ex

```diff
@@ -145,6 +145,38 @@ defmodule Gnat.Jetstream.API.Object do
     end
   end
 
+  @doc """
+  Returns true if the provided stream is an Object bucket, false otherwise
+  ## Parameters
+  * `stream_name` - the stream name to test
+  """
+  @spec is_object_bucket_stream?(stream_name :: binary()) :: boolean()
+  def is_object_bucket_stream?(stream_name) do
+    String.starts_with?(stream_name, "OBJ_")
+  end
+
+  @doc """
+  Returns a list of all Object buckets
+  """
+  @spec list_buckets(conn :: Gnat.t()) :: {:error, term()} | {:ok, list(String.t())}
+  def list_buckets(conn) do
+    with {:ok, %{streams: streams}} <- Stream.list(conn) do
```

Comment:
The `Stream.list` function accepts optional arguments to fetch additional pages of streams. It looks like both `Object.list_buckets` and `KV.list_buckets` only look for matching stream names in the first page of stream names. Maybe we should accept optional arguments to look at the next page? I'd be interested to know how this is handled by the cli.

Reply:
@mmmries good callout.  I had not considered this.  I do agree that this can be handled in a separate PR.  I have some thoughts below on how we could accomplish this.  Let me know what you think.

CLI implements list buckets [here](https://github.com/nats-io/natscli/blob/0fc79e766be8ea67c07b13c869563f1b8e30b848/cli/object_command.go#L348).  I think that listing streams is implemented [here](https://github.com/nats-io/natscli/blob/0fc79e766be8ea67c07b13c869563f1b8e30b848/cli/stream_command.go#L748).

I can think of a couple of scenarios to consider.

* `list_buckets` in both modules adhere to the original contract of `Stream.list` (`{:ok, %{total: _, offset: 0, limit: 1024, streams: _}}`).  This would be a breaking change to KV's `list_buckets`.
* `list_buckets` for both KV and Object have their own contract that looks like `{:ok, ["TEST_BUCKET_1", "TEST_BUCKET_2"], %{total:_, offset: 0, limit: 1024}}`.  `list_buckets/2` in either module will optionally accept a parameter which specifies the offset and limit of the next page.

I think that a breaking change should be avoided.  `list_bucket/2` can expose the page API to the client and let the client handle pagination afterwards.